packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -93,6 +103,15 @@ export interface MessageManagerState<M extends AbstractMessage>
   unapprovedMessagesCount: number;
 }
 
+/**
+ * Represents the parameters to pass to the signing method once the signature request is approved.
+ */
+export type MessageParams =
```

Comment:
This is a circular dependency since the concrete classes need the abstract but the abstract now references the children.

Ideally, we want the abstract to know nothing about any child classes.

Can we not just use `AbstractMessageParams` as by definition, it should include all the common properties?